lib/winapi/libloaderapi.c

```diff
@@ -0,0 +1,37 @@
+#include <libloaderapi.h>
+#include <winbase.h>
+#include <windef.h>
+#include <winerror.h>
+#include <xboxkrnl/xboxkrnl.h>
+
+#include <assert.h>
+#include <stdlib.h>
+
+HMODULE LoadLibraryExA (LPCSTR lpLibFileName, HANDLE hFile, DWORD dwFlags)
+{
+    assert(hFile == NULL);
+    assert(dwFlags == 0);
+
+    // Always fail with not having found the library
+    SetLastError(ERROR_MOD_NOT_FOUND);
+    return NULL;
+}
+
+HMODULE LoadLibraryA (LPCSTR lpLibFileName)
+{
+    return LoadLibraryExA(lpLibFileName, NULL, 0);
+}
+
+BOOL FreeLibrary (HMODULE hLibModule)
+{
+    assert(hLibModule != NULL);
+
+    // Always claim success when free'ing a library
+    return TRUE;
+}
+
+FARPROC GetProcAddress (HMODULE hModule, LPCSTR lpProcName)
+{
+    // Always fail with not having found the export
```

Comment:
nit: This could `assert(hModule != NULL)` according to the MSDN (because that doesn't list `NULL` as valid input).

However, I believe I have actually seen DLLs use `GetProcAddress(NULL, "symbol")` to find their own symbol?
Would be interesting if someone could check on Windows.

Reply:
I just checked that and it indeed works like this on Windows (even works in normal exe files for functions marked as `dllexport`).

Reply:
I added a comment about this, please verify I described the behaviour correctly.

Reply:
Yeah it's fine.